src/main/java/upb/dice/twitter/TweetSearchApplication.java

```diff
@@ -1,156 +1,77 @@
-        Map<List<String>, String> pairStringMap = new HashMap<>();
-        try (BufferedReader br = new BufferedReader(new FileReader("Latitude_Longitude_MaxID.txt"))) {
-            String line;
-            while ((line = br.readLine()) != null) {
-                String[] splitString = line.split(":");
-                String latitude = splitString[0];
-                String longitude = splitString[1];
-                String locMaxId = splitString[2];
-                pairStringMap.put(Arrays.asList(latitude, longitude), locMaxId);
-            }
-            String lon_str = String.valueOf(geoLocation.getLatitude());
-            String lat_str = String.valueOf(geoLocation.getLatitude());
-            if (pairStringMap.containsKey(Arrays.asList(lon_str, lat_str))) {
-                String id_string = pairStringMap.get(Arrays.asList(lon_str, lat_str));
-                return Long.parseLong(id_string);
-            }
-            return 1L;
-        }
-    }
-
-
-    private void queryBuild(String key) throws TwitterException, IOException {
-        long key_id_1 = keyParse(key);
-        Query query1 = new Query(key);
-        if (key_id_1 != 1) {
-            query1.sinceId(key_id_1);
-        }
-        getTweet(query1);
-    }
-
-    //Location based Tweet Search
-    private void queryBuild(double latitude, double longitude, double radius) throws TwitterException, IOException {
-        GeoQuery geoQuery = new GeoQuery(new GeoLocation(latitude, longitude));
-        Query query = new Query();
-        query.setGeoCode(geoQuery.getLocation(), radius, Query.KILOMETERS);
-        lat = latitude;
-        lon = longitude;
-
-//       query.until("2019-02-31").since("2019-02-01");   //Specific time period
-        long key_id_1 = locationParse(new GeoLocation(latitude, longitude));
-        if (key_id_1 != 1) {
-            query.sinceId(key_id_1);
-        }
-        getTweet(query);
-    }
-
-
-    //Extract the tweets based on the query built
-    private void getTweet(Query query) throws TwitterException, IOException {
-        QueryResult queryResult;
-        System.out.println("Tweets search Start");
-        FileWriter writer = new FileWriter("Tweets.txt", true);
-        int counter = 0;
-        do {
-            queryResult = twitter.search(query);
-            List<Status> queryTweets = queryResult.getTweets();
-            long maxId;
-            if (counter == 0) {
-                maxId = queryResult.getMaxId();
-                //Location based
-                if (query.getQuery() == null) {
-                    try (BufferedWriter out = new BufferedWriter(new FileWriter("Latitude_Longitude_MaxID.txt", true))) {
-                        out.append(String.valueOf(lat)).append(":").append(String.valueOf(lon)).append(":").append(String.valueOf(maxId));
-                        out.append('\n');
+        CommandLine finalCmd = cmd;
+        assert finalCmd != null;
```

Comment:
I would suggest to use an `if` with an error message instead of the `assert`.
Typically, `asserts` shouldn't be used in the final code.